x/examples/website-wrapper-app/wrapper_app_project/.scripts/build.mjs

```diff
@@ -95,14 +99,33 @@ export default async function main(
       additionalDomains = [additionalDomains];
     }
 
+    const entryUrlParts = new URL(entryUrl);
+
+    const entryDomain = entryUrlParts.hostname;
+
+    const [entryTld, ...entryDomainParts] = entryDomain.split(".").reverse();
+
+    // Infer an app name from the base entry domain part 
+    const defaultAppName = entryDomainParts[0].toLocaleLowerCase().replaceAll(
+      /\w[a-z0-9]*[-_]*/g,
+      match => {
+        match = match.replace(/[-_]+/, " ");
+
+        return match.charAt(0).toUpperCase() + match.slice(1).toLowerCase();
+      }
+    )
+  
+    // Infer an app ID from the entry domain by reversing it (e.g. `www.example.com` becomes `com.example.www`)
+    const defaultAppId = entryTld + "." + entryDomainParts.join(".").replaceAll("-", "").toLocaleLowerCase();
+
     fs.writeFileSync(
       destinationFilepath.replace(/\.handlebars$/, ""),
       template({
         platform,
         entryDomain,
-        navigationUrl: navigationUrl
-          ? navigationUrl
-          : `https://${entryDomain}/`,
+        appId: appId ?? defaultAppId,
```

Comment:
This is a bit confusing and can lead to unexpected behavior. If someone specifies the appID, you should always use that (not try to generate one). It it's not specified, you should always generate/extract it, and fail if you can't.